auth/src/main/java/com/firebase/ui/auth/ui/email/RegisterEmailFragment.java

```diff
@@ -210,6 +213,39 @@ public void onClick(View view) {
         });
     }
 
+    private void setUpPrivacyPolicy() {
+        if (TextUtils.isEmpty(mHelper.getFlowParams().privacyPolicyUrl)) {
+            return;
+        }
+
+        ForegroundColorSpan foregroundColorSpan =
+                new ForegroundColorSpan(ContextCompat.getColor(getContext(), R.color.linkColor));
+
+        String link = getString(R.string.privacy_policy);
+        SpannableStringBuilder spannableStringBuilder = new SpannableStringBuilder(link);
+        int start = mAgreementText.length();
+        spannableStringBuilder.setSpan(foregroundColorSpan, start, start + link.length(), 0);
+
+        mAgreementText.append(" and the ");
+        mAgreementText.append(spannableStringBuilder);
+        mAgreementText.setOnClickListener(new View.OnClickListener() {
```

Comment:
Yeah, that's what I did. Since you mentioned having issues with the Spans, I went ahead and created a patch for you: [Fix_various_bugs.zip](https://github.com/firebase/FirebaseUI-Android/files/1030730/Fix_various_bugs.zip).

If you want to give it a go yourself, hack at it! 😄 Otherwise, just extract the zip and then apply the patch in Android Studio by calling up the shortcut: `Ctrl/Cmd + Shift + A` -> "Apply patch".